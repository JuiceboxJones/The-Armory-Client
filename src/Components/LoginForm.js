import React, { useState } from 'react';
import TokenService from '../services/token-service';
import AuthApiService from '../services/auth-api-service';

export default function LoginForm(props) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('')

  function handleLogin (e) {
      e.preventDefault();
      AuthApiService.postLogin({email, password})
      .then(res => {
        TokenService.saveAuthToken(res.authToken);
        props.onLoginSuccess();
      })
      .catch (error =>
      setError(error)
      )
  };


  return (
    <div>
      <h2>Login</h2>
      <p>{error.error}</p>
      <form className="LoginForm" onSubmit={e => handleLogin(e)}>
        <label htmlFor="email-input">Email or Username: </label>
        <input
          required
          type='text'
          name="email-input"
          id="email-input"
          placeholder="username@example.com"
          onChange={e => setEmail(e.target.value)}
        />
        <label htmlFor="password-input">Password: </label>
        <input
          required
          type="password"
          name="password-input"
          id="password-input"
          onChange={e => setPassword(e.target.value)}
        />
        <button type="submit" className="submit-button">
        Login 
        </button>
      </form>
    </div>
  );
}
