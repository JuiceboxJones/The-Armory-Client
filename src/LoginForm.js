import React, {useState, useEffect} from 'react';
import { Link } from 'react-router-dom';

export default function LoginForm(props) {

  const {email, setEmail} = useState('');
  const {password, setPassword} = useState('')

  const handleSubmit = e => {
    e.preventDefault();
    //send (email, password) to authenication endpoint
  };

  return(
    <div>
      <h2>Login</h2>
      <form className='LoginForm' onSubmit={e => handleSubmit(e)}>
      <label htmlFor='email-input'>Email:</label>
      <input 
      required 
      type='email' 
      name='email-input' 
      id='email-input'
      placeholder='username@example.com'
      onChange={e => setEmail({ email: e.target.value })}/>
      <label htmlFor='password-input'>Password:</label>
      <input 
      required 
      type='password' 
      name='password-input' 
      id='password-input'
      onChange={e => setPassword({ password: e.target.value })}/>
      <input type='submit' />
      </form>
      <Link to= '/(registration route name)'>Not a member? Sign up here.</Link>
    </div>
  )

}