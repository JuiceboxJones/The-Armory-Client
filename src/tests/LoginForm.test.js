import React from 'react';
import LoginForm from '../Components/LoginForm';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';

describe('LoginForm component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <LoginForm />
      </MemoryRouter>, div
    );
    ReactDOM.unmountComponentAtNode(div);
  })
})