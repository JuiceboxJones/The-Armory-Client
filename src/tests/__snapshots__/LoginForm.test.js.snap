// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the UI as expected 1`] = `
<div>
  <h2>
    Login
  </h2>
  <p />
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="email-input"
    >
      Email or Username: 
    </label>
    <input
      id="email-input"
      name="email-input"
      onChange={[Function]}
      placeholder="username@example.com"
      required={true}
      type="text"
    />
    <label
      htmlFor="password-input"
    >
      Password: 
    </label>
    <input
      id="password-input"
      name="password-input"
      onChange={[Function]}
      required={true}
      type="password"
    />
    <button
      className="submit-button"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
